receiver/receivercreator/receiver.go

```diff
@@ -0,0 +1,173 @@
+	viperConfig := viper.New()
+	viperConfig.Set(staticSubConfig.fullName, map[string]interface{}{})
+	subreceiverConfig := viperConfig.Sub(staticSubConfig.fullName)
+
+	// Merge in the config values specified in the config file.
+	if err := subreceiverConfig.MergeConfigMap(staticSubConfig.config); err != nil {
+		return nil, fmt.Errorf("failed to merge subreceiver config from config file: %v", err)
+	}
+
+	// Merge in subConfigFromEnv containing values discovered at runtime.
+	if err := subreceiverConfig.MergeConfigMap(subConfigFromEnv); err != nil {
+		return nil, fmt.Errorf("failed to merge subreceiver config from discovered runtime values: %v", err)
+	}
+
+	receiverConfig, err := config.LoadReceiver(staticSubConfig.fullName, viperConfig, factories)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load subreceiver config: %v", err)
+	}
+	// Sets dynamically created receiver to something like receiver_creator/1/redis{endpoint="localhost:6380"}.
+	// TODO: Need to make sure this is unique (just endpoint is probably not totally sufficient).
+	receiverConfig.SetName(fmt.Sprintf("%s/%s{endpoint=%q}", dr.cfg.Name(), staticSubConfig.fullName, subreceiverConfig.GetString("endpoint")))
+	return receiverConfig, nil
+}
+
+// createRuntimeReceiver creates a receiver that is to discovered at runtime
+func (dr *receiverCreator) createRuntimeReceiver(cfg configmodels.Receiver) (receiver.MetricsReceiver, error) {
+	factory, err := factories.get(cfg.Type())
+	if err != nil {
+		return nil, err
+	}
+	receiverFactory := factory.(receiver.Factory)
+
+	return receiverFactory.CreateMetricsReceiver(dr.logger, cfg, dr)
+}
+
+// Start receiver_creator.
+func (dr *receiverCreator) Start(host component.Host) error {
+	// TODO: Temporarily load a single instance of all subreceivers for testing.
+	// Will be done in reaction to observer events later.
+	for _, subconfig := range dr.cfg.subreceiverConfigs {
+		cfg, err := dr.loadRuntimeReceiverConfig(subconfig, map[string]interface{}{})
+		if err != nil {
+			return err
+		}
+		recvr, err := dr.createRuntimeReceiver(cfg)
+		if err != nil {
+			return err
+		}
+
+		if err := recvr.Start(host); err != nil {
+			return fmt.Errorf("failed starting subreceiver %s: %v", cfg.Name(), err)
+		}
+
+		dr.receivers = append(dr.receivers, recvr)
+	}
+
+	// TODO: Can result in some receivers left running if an error is encountered
+	// but starting receivers here is only temporary and will be removed when
+	// observer interface added.
+
+	return nil
+}
+
+// Shutdown stops the receiver_creator and all its receivers started at runtime.
+func (dr *receiverCreator) Shutdown() error {
+	var errs []string
+
+	for _, recvr := range dr.receivers {
+		if err := recvr.Shutdown(); err != nil {
+			// TODO: Should keep track of which receiver the error is associated with
+			// but require some restructuring.
+			errs = append(errs, err.Error())
+		}
+	}
+
+	if len(errs) > 0 {
```

Comment:
We typically use `oterr.CombineErrors` when there are multiple errors to return.